src/pipx/venv.py

```diff
@@ -241,10 +232,24 @@ def install_package_no_deps(self, package_or_url: str, pip_args: List[str]) -> s
 
     def get_venv_metadata_for_package(self, package: str) -> VenvMetadata:
         data = json.loads(
-            get_script_output(
-                self.python_path, VENV_METADATA_INSPECTOR, package, self.bin_path
-            )
+            run_subprocess(
+                [
+                    self.python_path,
+                    VENV_METADATA_INSPECTOR_FILE,
+                    package,
+                    self.bin_path,
+                ],
+                capture_stderr=False,
+            ).stdout
```

Comment:
Would this work if pipx is run in a non-flat context? e.g. zipapp or py2exe.

Reply:
The code here is exactly what used to be there inside of `get_script_output()`, the only exception being no longer using the `-c` switch after the python interpreter (i.e. submitting the name of the script to the python interpreter instead of the entire string contents of the script.)  So I assume it should work the same way.

Is there an example of a zipapp or py2exe that I can test to double-check?

Reply:
The problem here is the usage of `__file__`, which is [not available for py2exe](https://stackoverflow.com/questions/24140593) (and other bundling utilities, since they generate do no maintain the source code tree).

A simple example with zipapp:

```python
# foo/__main__.py
import subprocess
import sys

from mod import __file__ as mod_file

subprocess.check_call([sys.executable, mod_file])
```

```python
# foo/mod.py
if __name__ == "__main__":
    print("Executing mod")
```

```console
$ python3 foo/__main__.py   # This works.
Executing mod
```

```console
$ python3 -m zipapp foo
$ python3 foo.pyz  # But this does not.
python3: can't find '__main__' module in 'foo.pyz/mod.py'
Traceback (most recent call last):
  File "lib/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "lib/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "foo.pyz/__main__.py", line 6, in <module>
  File "lib/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['python3', 'foo.pyz/mod.py']' returned non-zero exit status 1.
```

Reply:
So to make sure I understand, this would be a problem if we bundle `pipx` itself, not so much a problem for the client apps?

Reply:
Yes, that is correct.

Reply:
OK, I reverted back to using the text of `pipx_metadata_inspector.py` as before.